packages/client/src/Config.ts

```diff
@@ -50,64 +50,241 @@ export interface EthereumNetworkConfig {
 export interface StreamrClientConfig {
     /** Custom human-readable debug id for client. Used in logging. */
     id?: string
+
+    /**
+     * Override the default logging level.
+     */
     logLevel?: LogLevel
+
     /**
-    * Authentication: identity used by this StreamrClient instance.
-    * Can contain member privateKey or (window.)ethereum
+    * The Ethereum identity to be used by the client. Either a private key
+    * or a window.ethereum object.
     */
     auth?: PrivateKeyAuthConfig | ProviderAuthConfig
 
-    /** Attempt to order messages */
+    /**
+     * Set to true to enable message ordering.
+     *
+     * Due to the distributed nature of the network, messages may occasionally
+     * arrive to the client out-of-order. By enabling message ordering, the
+     * client will reorder received messages to the intended order for you.
+     *
+     * */
     orderMessages?: boolean
+
+    /**
+     * Set to true to enable gap filling.
+     *
+     * Some messages may occasionally not reach the client due to networking
+     * issues. Missing messages form gaps that are often detectable and
+     * retrievable on demand. By enabling gap filling, the client will detect
+     * and fix gaps automatically for you.
+     */
     gapFill?: boolean
+
+    /**
+     * When gap filling is enabled, this option controls the maximum amount of
+     * times a gap will try to be actively filled before giving up and
+     * proceeding forwards.
+     */
     maxGapRequests?: number
-    retryResendAfter?: number
+
+    /**
+     * When gap filling is enabled and a gap is encountered, this option
+     * defines the amount of time to wait before attempting to _actively_ fill
+     * in the gap.
+     *
+     * Rationale: data may just be arriving out-of-order and the missing
+     * message(s) may be on their way. For efficiency, it makes sense to wait a
+     * little before actively attempting to fill in a gap, as this involves
+     * a resend request / response interaction with a storage node.
+     */
     gapFillTimeout?: number
 
     /**
-     * Message encryption/decryption
+     * When gap filling is enabled and a gap is encountered, a resend request
```

Comment:
This is not always done, only after `gapFillTimeout`. 

We could say something like "If we make a storage node request in order to fill a gap, and the request fails, the client can retry the fetch from (another?) storage. This option controls how long to wait for a storage node response from the storage node before proceeding to the next attempt.

The number is in milliseconds."

(The unit of time, i.e. milliseconds, could be included in each of the sections where applicable: `retryResendAfter`, `keyRequestTimeout`, `newWebrtcConnectionTimeout`).

Reply:
Changed wording to emphasize conditionality of resend request:
> When gap filling is enabled and a gap is encountered, a resend request
_may eventually_ be sent ...

Reply:
I would leave out the details of requesting from another storage node etc.

Reply:
Added units to descriptions